### ISSUE TITLE
Performance problems with a small elementary-abelian-by-cyclic group

### ISSUE THREAD

**Issue by Stefan-Kohl** [gap-system/gap#3523]
### Observed behaviour

gap> H := GL(5,5);
GL(5,5)
gap> N := GF(5)^5;
( GF(5)^5 )
gap> repeat h := PseudoRandom(H); until Order(h) mod (11\*71) = 0;
gap> h := h^(Order(h)/(11\*71));;
gap> G := SemidirectProduct(Group(h),N); # a group of type 5^5 ><| (11*71)
<matrix group of size 2440625 with 2 generators>
gap> P := Action(G,GF(5)^6,OnRight); # immediate ...
<permutation group with 2 generators>
gap> time;
240
gap> phi := IsomorphismPermGroup(G); 

  ... hangs, or takes at least MUCH too long ...

gap> phi := IsomorphismPcGroup(G);

  ... hangs, or takes at least MUCH too long ...

gap> SmallerDegreePermutationRepresentation(P);

  ... hangs, or takes at least MUCH too long ...

### Copy and paste GAP banner (to tell us about your setup)

 ┌───────┐   GAP 4.10.2 of 19-Jun-2019
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv3
 Configuration:  gmp 5.1.3
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.2, PrimGrp 3.3.2, SmallGrp 1.3, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

**Comment by hulpke** [comment]
With #3494:

    gap>  P := Action(G,GF(5)^6,OnRight);time;
     <permutation group with 2 generators>
    84
    gap> phi := IsomorphismPermGroup(G);;time;
    96
    gap> sma:=SmallerDegreePermutationRepresentation(P:cheap);;time;
    73

**Comment by wilfwilson** [comment]
@Stefan-Kohl given that #3494 is now merged, would you be happy for this issue to be closed? Or are you still unhappy that `SmallerDegreePermutationRepresentation(P);` (without the cheap option) takes a long time?

**Comment by Stefan-Kohl** [comment]
@wilfwilson I think the performance problem with IsomorphismPcGroup has not been addressed so far. -- As representing the given group as a polycyclic group is likely best for most computations, I think this is the main issue here. -- After resolving this as well, the issue can be closed.

**Comment by wilfwilson** [comment]
@Stefan-Kohl for me, the `IsomorphismPcGroup` line is quick now, after #3494 is merged:
```
...
gap> G := SemidirectProduct(Group(h),N);
<matrix group of size 2440625 with 2 generators>
gap> P := Action(G,GF(5)^6,OnRight);
<permutation group with 2 generators>
gap> phi := IsomorphismPcGroup(G);; time;
78
```
Is it not so for you?

**Comment by Stefan-Kohl** [comment]
@wilfwilson At the moment I cannot easily try it with #3494 merged -- but just with the released version (4.10.2). -- In any case, if #3494 resolves also the problem with IsomorphismPcGroup, then the issue can be closed.

**Comment by wilfwilson** [comment]
I've just checked out a commit from before #3494 was merged: it was indeed slow then, but it's definitely fast now in `master`. So I'll close 🙂 Thanks @hulpke.
